### ISSUE TITLE
compiler error installing MonetDBlite 0.5.0 on Ubuntu 17.10

### ISSUE THREAD

**Issue by rgayler** [hannesmuehleisen/MonetDBLite-R#10]
Hi,

I upgraded Ubuntu to 17.10 recently and installed  MonetDBLite 0.4.1 with no problem.
Today, attempting to install 0.5.0 from within RStudio I get a compiler error:

```
> install.packages("MonetDBLite")
Installing package into ‘/home/ross/R/x86_64-pc-linux-gnu-library/3.4’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/src/contrib/MonetDBLite_0.5.0.tar.gz'
Content type 'application/x-gzip' length 1381343 bytes (1.3 MB)
==================================================
downloaded 1.3 MB

* installing *source* package ‘MonetDBLite’ ...
** package ‘MonetDBLite’ successfully unpacked and MD5 sums checked
** libs
gcc -std=gnu99  -g -O2 -fdebug-prefix-map=/build/r-base-8tMiyU/r-base-3.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -I/usr/share/R/include -DNDEBUG -fpic -Wno-format -DLIBGDK -DLIBMAL -DLIBOPTIMIZER -DLIBSTREAM -DHAVE_EMBEDDED_R -DMONETDBLITE_COMPILE -Imonetdblite/src/ -Imonetdblite/src/common -Imonetdblite/src/embedded -Imonetdblite/src/gdk -Imonetdblite/src/mal/mal -Imonetdblite/src/mal/modules -Imonetdblite/src/mal/optimizer -Imonetdblite/src/mal/sqlbackend -Imonetdblite/src/sql/include -Imonetdblite/src/sql/common -Imonetdblite/src/sql/server -Imonetdblite/src/sql/storage -Imonetdblite/src/sql/storage/bat -MMD -MF build/deps/monetdblite/common/stream.d -c monetdblite/src/common/stream.c -o build/objects/monetdblite/common/stream.o
cc1: error: -Wformat-security ignored without -Wformat [-Werror=format-security]
cc1: some warnings being treated as errors
Makevars:194: recipe for target 'build/objects/monetdblite/common/stream.o' failed
make: *** [build/objects/monetdblite/common/stream.o] Error 1
ERROR: compilation failed for package ‘MonetDBLite’
* removing ‘/home/ross/R/x86_64-pc-linux-gnu-library/3.4/MonetDBLite’
* restoring previous ‘/home/ross/R/x86_64-pc-linux-gnu-library/3.4/MonetDBLite’
Warning in install.packages :
  installation of package ‘MonetDBLite’ had non-zero exit status

The downloaded source packages are in
	‘/tmp/Rtmp7rYzho/downloaded_packages’
```
I don't have a Makevars file, so presumably the compiler flags are the defaults for installation. I have so far avoided tinkering with the compiler flags as I don't know what the maintainers would regard as essential.

Is this something the maintainers would care about or more likely something idiosyncratic about my set-up?

**Comment by hannesmuehleisen** [comment]
Hi whats the output of sessionInfo() thanks

**Comment by rgayler** [comment]
Thanks - that was rapid!

```
> sessionInfo()
R version 3.4.2 (2017-09-28)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 17.10

Matrix products: default
BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C               LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8    LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.4.2 tools_3.4.2    yaml_2.1.14   
```

**Comment by hannesmuehleisen** [comment]
that looks fine, and what's the output of `R CMD config CFLAGS`

**Comment by rgayler** [comment]
```
ross@ross-ThinkPad-X1-Carbon-2nd:~$ R CMD config CFLAGS
-g -O2 -fdebug-prefix-map=/build/r-base-8tMiyU/r-base-3.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g

```

**Comment by hannesmuehleisen** [comment]
Yes those are indeed conflicting flags. I suggest to put something like
`CFLAGS = -g -O2` into ~/.R/Makevars

**Comment by rgayler** [comment]
Thanks. I will try that after dinner and let you know the outcome.

**Comment by rgayler** [comment]
Thanks! That worked perfectly.

I am not aware of having explicitly set those conflicting flags, so I assume they must be the defaults for the current installation of R on my platform.

I know almost nothing about the package installation process. I only knew about ~/.R/Makevars because I had to create it to install the rstan package - there were instructions on the rstan repo because of a temporary build issue on some platforms. I removed that Makevars file after I first encountered the MonetDBLite problem, to see whether that was the cause.

So, is `CFLAGS = -g -O2` something I can safely leave permanently in ~/.R/Makevars or should I have a MonetDBLite-specific Makevars file that i only use when installing MonetDBLite?

Thanks
